components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -152,22 +152,44 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
+                String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
                 try {
                     OMElement payLoad,body;
                     body = messageContext.getEnvelope().getBody();
                     payLoad = body.getFirstChildWithName(new QName(RFCConstants.BAPIRFC));
                     log.info("Received RFC/Meta DATA: " + payLoad);
                     String rfcFunctionName = RFCMetaDataParser.getBAPIRFCFucntionName(payLoad);
                     log.info("Looking up the BAPI/RFC function: " + rfcFunctionName + ". In the " +
-                             "meta data repository");
-
-                    JCoFunction function = getRFCfunction(destination, rfcFunctionName);
-                    RFCMetaDataParser.processMetaDataDocument(payLoad, function);
-                    String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
-                    String responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                            "meta data repository");
+                    String responseXML;
+                    if (isTransaction(messageContext)) {
+                        JCoContext.begin(destination);
+                        log.info("Begin transaction.");
+                        JCoFunction function = getRFCfunction(destination, rfcFunctionName);
+                        RFCMetaDataParser.processMetaDataDocument(payLoad, function);
+                        responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                        JCoFunction commitFunction = getRFCfunction(destination, SAPConstants.BAPI_TRANSACTION_COMMIT);
+                        evaluateRFCfunction(commitFunction, destination, escapeErrorHandling);
+                        log.info("Commit transaction.");
+                    } else {
+                        JCoFunction function = getRFCfunction(destination, rfcFunctionName);
+                        RFCMetaDataParser.processMetaDataDocument(payLoad, function);
+                        responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                    }
                     processResponse(messageContext, responseXML);
                 } catch (Exception e) {
+                    if (isTransaction(messageContext)) {
+                        JCoFunction rollbackFunction = getRFCfunction(destination,
+                                SAPConstants.BAPI_TRANSACTION_ROLLBACK);
+                        evaluateRFCfunction(rollbackFunction, destination, escapeErrorHandling);
+                        log.info("Rollback transaction.");
+                    }
                     sendFault(messageContext, e , SAP_TRANSPORT_ERROR);
+                } finally {
+                    if (isTransaction(messageContext)) {
+                        JCoContext.end(destination);
+                        log.info("End transaction.");
```

Comment:
I think debug logs will be more appropriate for the commit, rollback and end operations. WDYT?

Reply:
Yes. We need to move to debug level. Otherwise, there will be some unnecessary logs.

Reply:
+1. Fixed.